src/QueryEditor.tsx

```diff
@@ -1,16 +1,31 @@
-import defaults from 'lodash/defaults';
+import { defaults } from 'lodash';
 
-import React, { ChangeEvent, PureComponent } from 'react';
+import React, { ChangeEvent, PureComponent, SyntheticEvent } from 'react';
 import { LegacyForms } from '@grafana/ui';
 import { QueryEditorProps } from '@grafana/data';
 import { DataSource } from './datasource';
 import { defaultQuery, MyDataSourceOptions, MyQuery } from './types';
+import { config } from '@grafana/runtime';
 
-const { FormField } = LegacyForms;
+const { FormField, Switch } = LegacyForms;
 
 type Props = QueryEditorProps<DataSource, MyQuery, MyDataSourceOptions>;
 
 export class QueryEditor extends PureComponent<Props> {
+  supportsStream = true;
+
+  componentDidMount() {
+    const version = config.buildInfo.version;
+    if (version.startsWith('7.')) {
```

Comment:
Since this adds more craft to the starter - maybe we can simply put `Grafana v8+` notice to a switch label? I think we can provide this code as part of streaming tutorial doc... But I think we really should not deal with versioning in a starter.

Reply:
pushed